components/Auth/Login.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+  Button,
+  Divider,
+  HStack,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Text,
+  VStack,
+} from '@chakra-ui/react'
+import { yupResolver } from '@hookform/resolvers/yup'
+import React, { Dispatch, SetStateAction, useState } from 'react'
+import { Controller, ErrorOption, useForm } from 'react-hook-form'
+import { emailSignIn } from 'utils/firebase'
+import * as yup from 'yup'
+import { GoogleSignInComponent } from '.'
+
+interface LoginProps {
+  setCurrentState: Dispatch<SetStateAction<'sign_up' | 'log_in'>>
+}
+
+interface LoginForm {
+  email: string
+  password: string
+}
+
+const loginFormSchema = yup.object({
+  email: yup.string().required().email(),
+  password: yup.string().required(),
+})
+
+const initialState = {
+  email: '',
+  password: '',
+}
+
+const Login = ({ setCurrentState }: LoginProps): JSX.Element => {
+  const methods = useForm({
+    defaultValues: initialState,
+    resolver: yupResolver(loginFormSchema),
+  })
+
+  const [showPassword, setShowPassword] = useState(false)
+
+  const onSubmit = ({ email, password }: LoginForm) => {
+    emailSignIn(email, password)
+      .then((res) => {
+        // TODO handle different types of possible firebase errors and return types also add a toast
+        if (res.code !== 200) {
+          const error: ErrorOption = {
+            message: res.message.substring(9),
+          }
+          methods.setError('password', error)
+        }
+      })
+      .catch((err) => console.log(err))
+  }
+
+  return (
+    <VStack flex={1} w="100%" justifyContent="space-between" color="black">
+      <Text fontSize={'3xl'} fontWeight="bold">
+        Login
+      </Text>
+      <GoogleSignInComponent />
+      <HStack w="100%">
+        <Divider border="1px solid gray.400" />
+        <Text> or </Text>
+        <Divider border="1px solid gray.400" />
+      </HStack>
+
+      <VStack w="100%" maxW="xs" as="form">
+        <Controller
```

Comment:
Any reason we're using React Hook Form's `Controller` instead of [`register`](https://react-hook-form.com/api/useform/register)? [Chakra's `Input` exposes its ref](https://github.com/MutualAidNYC/services-lists/blob/main/components/Forms/Input.tsx#L38) and we have [a helper hook](https://github.com/MutualAidNYC/services-lists/blob/main/hooks/useHookFormProps.ts) for registering stuff to React Hook Form.

This React Hook Form stuff is based off of work code awhile ago though so code probably do with some refactoring (in a separate PR).

Reply:
Ill switch it over :)